http-gateways/PATH_GATEWAY.md

```diff
@@ -182,10 +184,10 @@ For example:
 - [application/vnd.ipld.raw](https://www.iana.org/assignments/media-types/application/vnd.ipld.raw) – disables [IPLD codec deserialization](https://ipld.io/docs/codecs/), requests a verifiable raw [block](https://docs.ipfs.io/concepts/glossary/#block) to be returned
 - [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car) – disables [IPLD codec deserialization](https://ipld.io/docs/codecs/), requests a verifiable [CAR](https://docs.ipfs.io/concepts/glossary/#car) stream to be returned
 - [application/x-tar](https://en.wikipedia.org/wiki/Tar_(computing)) – returns UnixFS tree (files and directories) as a [TAR](https://en.wikipedia.org/wiki/Tar_(computing)) stream. Returned tree starts at a root item which name is the same as the requested CID. Produces 400 Bad Request for content that is not UnixFS.
-<!-- TODO: https://github.com/ipfs/go-ipfs/issues/8823
-- application/vnd.ipld.dag-json OR application/json – requests IPLD Data Model representation serialized into [DAG-JSON format](https://ipld.io/docs/codecs/known/dag-json/)
-- application/vnd.ipld.dag-cbor OR application/cbor - requests IPLD Data Model representation serialized into [DAG-CBOR format](https://ipld.io/docs/codecs/known/dag-cbor/)
--->
+- [application/vnd.ipld.dag-json](https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-json) – requests [IPLD Data Model](https://ipld.io/docs/data-model/) representation serialized into [DAG-JSON format](https://ipld.io/docs/codecs/known/dag-json/). If the requested CID already has `dag-json` (0x0129) codec, data is validated as DAG-JSON before being returned as-is. Invalid DAG-JSON produces HTTP Error 500.
+- [application/vnd.ipld.dag-cbor](https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-cbor) – requests [IPLD Data Model](https://ipld.io/docs/data-model/) representation serialized into [DAG-CBOR format](https://ipld.io/docs/codecs/known/dag-cbor/). If the requested CID already has `dag-cbor` (0x71) codec,  data is validated as DAG-CBOR before being returned as-is. Invalid DAG-CBON produces HTTP Error 500.
+- [application/json](https://www.iana.org/assignments/media-types/application/json) – same as `application/vnd.ipld.dag-json`, unless the CID's codec already is `json` (0x0200). Then, the raw JSON block can be returned.
+- [application/cbor](https://www.iana.org/assignments/media-types/application/cbor) – same as `application/vnd.ipld.dag-cbor`, unless the CID's codec already is `cbor` (0x51). Then, the raw CBOR block can be returned.
```

Comment:
> Issue found during the test rolout on `ipfs.io` gateways: https://github.com/protocol/bifrost-infra/issues/2290

:warning:  this comes with a potential footgun:  JSON  and CBOR stored as UnixFS files will now be returned as [IPLD's UnixFS Logical Format](https://ipld.io/specs/codecs/dag-pb/spec/#logical-format)  representation of a `raw` block, instead of JSON as-is with `application/json` header.

To illustrate: being able to do `?format=json`  on a directory is nice, but if we do it on a JSON file stored as UnixFS (`dag-pb` or `raw` block), then the response is JSON encoded as Base64 put inside of dag-json envelope:


**Expected:** 
```console
$ curl -v -H 'Accept: application/json' -H 'Content-type: application/json' https://bafkreie6koluvmeuiwbnpsjewa426mdbxyalygjjjvswqea2fuxpsuqc64.ipfs.dweb.link
{"name":"Claynosaurz #471","symbol":"DINO","descripti...
```

**Kubo 0.18.0-rc2:**
```console
$ curl -v -H 'Accept: application/json' -H 'Content-type: application/json' https://bafkreie6koluvmeuiwbnpsjewa426mdbxyalygjjjvswqea2fuxpsuqc64.ipfs.dweb.link
{"/":{"bytes":"eyJuYW1lIjoiQ2xheW5vc2F1cnogIzQ3MSIsInN5bWJvbCI6IkRJTk8iLCJkZXNjcmlwdGlvbiI6IkNsYXlub3NhdXJ6IG ...
```

We should not mangle JSON like this, but need to decide how to fix it on the implementation side,  then update the specification. (cc @hacdias)

Reply:
Some options:
1. detect when UnixFS node is a file, and do additional content-sniffing before deciding if node should be converted into IPLD Logical Format, skipping conversion if data is already in requested format (JSON or CBOR)
2. remove IPLD Logical Format conversion via `?format=json|cbor` – too many UX foot-guns. 
   - If we want dir listing as json, we would still have it under `?format=dag-json`.

Given unexpected consequences of our good UX intentions, 
I feel we should lean towards the latter (2): safer, makes smaller spec.

@hacdias if you feel the same, mind applying spec changes and opening a PR against Kubo that implements (2)? We need to fix this before 0.18 ships.

Reply:
I think **2** is the best solution. It took me quite longer than I wanted to make a PR (https://github.com/ipfs/kubo/pull/9565). I'm not sure if it's the best strategy, because we didn't discuss what we want to do  with conversions between JSON and CBOR for example. On the linked PR there's a table of what happens when you request each type.

Reply:
@lidel I've pushed a commit with some updates to remove the non-DAG versions.

Reply:
@hacdias mind restoring/adding basic info  about  `?format=json` and `?format=cbor`  and `Accept: application/json`  `Accept: application/cbor` being a way to fetch dag-json/dag-cbor with plain JSON/CBOR `Content-Type` for interop with other software that expect plain types?

Reply:
@lidel I reverted the commit I had pushed. I think what we had written is exactly what the current behavior is.